### PR TITLE
Improve README.md

### LINKED ISSUE #123: Need to improve README.md
Need a better README.md

### PR DESCRIPTION
**Pull request by jrcast**
Improve README.md

Closes #123

### CODE REVIEW

**Comment by shankar-forgerock** [comment]
To install the latest release secret-agent in a kubernetes environment, run:
may be better written as:
To install the latest release of  ``secret-agent`` in a kubernetes environment, run:
or
To install the latest ``secret-agent``  in a kubernetes environment, run:

**Comment by shankar-forgerock** [comment]
Thats a cool icon for secret-agent.

**Comment by shankar-forgerock** [comment]
Just to be easily followed, is the workflow 
1. Install secret-agent
2. Configure / Create a SAC configuration file
3. Push secret into the cluster.

**Comment by shankar-forgerock** [comment]
Does the cluster need to be running before 1. install or 2. configure?

**Comment by shankar-forgerock** [comment]
"Set up AWS Secret Manager" may be better titled "Set up Integration with AWS Secret Manager" or "To Use the secret-agent with AWS Secret Manager"

File: README.md [hunk 1]
```diff
@@ -1,47 +1,75 @@
...
+
+The `secret-agent` can be configured to back up all the generated secrets in a cloud provider's secret manager solution. When this feature is enabled, secrets stored in the secret managers are considered the source of truth.
+
+If a cloud provider has been configured, the operator will attempt to load the secret data from that cloud provider. If the secret is found in the cloud provider's secret manager, the operator will use the found data as the Kubernetes secret data. The operator will only generate new secrets if no secret data is found in the cloud provider.
+
+The `secret-agent` supports the following cloud providers:
+
+* Google Secret Manager
+* AWS Secrets Manager
+* ~~Azure Key Vault~~ (In progress)
```
Comment (maxres-fr) [inline README.md:51]: maybe link to issue #60

File: README.md [hunk 2]
```diff
@@ -77,11 +106,130 @@ spec:
...
+The secret generated by the this SAC would be stored as `devCluster-dev-ds-passwords-dirmanager-pw` in the AWS Secret Manager. The same name would apply to other cloud providers.
+
+Some key types require more than one secret to be backed up. Such key types require a public and private components stored separately. This is the case for key types: `ca`, `keypair`, `ssh` and `keytool`. These secrets use the same naming convention described above and append a suffix to the main name as constructed previously.
+
+Key Type | Name
+--- | ---
+`ca`      | $NAME-pem<br>$NAME-private-pem
+`keypair` | $NAME-pem<br>$NAME-private-pem
+`ssh`     | $NAME<br>$NAME-pub
+`keytool` | $NAME<br>$NAME-storepass<br>$NAME-keypass
```
Comment (maxres-fr) [inline README.md:156]: do we want to note the max size in AWS of 65k? or maybe below in some of the config?

**Review by maxres-fr — APPROVED** [review]

**Pull request merged**